subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/OutputScrapingExecutionResult.java

```diff
@@ -220,10 +224,39 @@ public ExecutionResult assertNotOutput(String expectedOutput) {
         return this;
     }
 
+    private static class LineWithDistance {
+        private final String line;
+        private final int distance;
+
+        public LineWithDistance(String line, int distance) {
+            this.line = line;
+            this.distance = distance;
+        }
+
+        public String getLine() {
+            return line;
+        }
+
+        public int getDistance() {
+            return distance;
+        }
+    }
+
     @Override
     public ExecutionResult assertContentContains(String actualText, String expectedOutput, String label) {
         String expectedText = LogContent.of(expectedOutput).withNormalizedEol();
         if (!actualText.contains(expectedText)) {
+            if (!expectedText.contains("\n")) {
+                Arrays.stream(actualText.split("\n"))
+                    // Measure Levenshtein distance for each line
+                    .map(line -> new LineWithDistance(line, StringUtils.getLevenshteinDistance(line, expectedText)))
+                    // Filter out lines that need more edits than half the length of the line
+                    .filter(pair -> pair.getDistance() < pair.getLine().length() / 2)
+                    // Find the closest match
+                    .min(Comparator.comparingInt(LineWithDistance::getDistance))
+                    .map(LineWithDistance::getLine)
+                    .ifPresent(similarLine -> failOnDifferentLine("Did not find expected text in " + label.toLowerCase() + ", found similar line.", expectedOutput, similarLine));
+            }
             failOnMissingOutput("Did not find expected text in " + label.toLowerCase() + ".", label, expectedOutput, actualText);
```

Comment:
Why should this treat single and multi line actual output differently?  Wouldn't the new format also be effective for single-line output?

Reply:
For simplicity. For multi-line comparisons the output would look pretty garbled, and falling back to the old format of just displaying the output looks fine to me. We can also improve this further in a new PR if we want to.

Reply:
Note that for multi-line outputs we'd need to also compare to similar-sized chunks of the output. Doing so might be worth it, but I don't have the time now to do that, too.